qemu/tests/s390x_cpu_topology_test.py

```diff
@@ -0,0 +1,41 @@
+from provider.cpu_utils import check_if_vm_vcpu_topology_match
+
+
+def run(test, params, env):
+    """
+    Check guest gets correct vcpu num, cpu cores, processors, sockets, siblings
+
+    1) Boot guest with options: -smp n,cores=x,threads=y,sockets=z...
+    2) Check cpu topology is expected as set up
+
+    :param test: QEMU test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    def check(p_name, exp, check_cmd):
+        """
+        Check the cpu property inside guest
+        :param p_name: Property name
+        :param exp: The expect value
+        :param check_cmd: The param to get check command
+        """
+        res = session.cmd_output(check_cmd).strip()
+        if int(res) != int(exp):
+            test.fail('The vcpu %s number inside guest is %s,'
+                      ' while it is set to %s' % (p_name, res, exp))
+
+    vm_name = params['main_vm']
+    os_type = params['os_type']
+    vm = env.get_vm(vm_name)
+    session = vm.wait_for_login()
+    if not check_if_vm_vcpu_topology_match(session, os_type, vm.cpuinfo,
+                                           test, vm.devices):
+        test.fail('CPU topology of guest is incorrect.')
+    vcpu_sockets = params['vcpu_sockets']
+    if params.get('check_core_id_cmd'):
+        for cpu_id in list(range(params['smp'])):
+            check('core_id', cpu_id, params['check_core_id_cmd'] % cpu_id)
+    if params.get('check_sockets_cmd'):
+        check('sockets', vcpu_sockets, params['check_sockets_cmd'])
```

Comment:
`one_cores_per_socket` and `two_cores_per_socket` here meet the same expectation of `vcpu_sockets == 2`.
How do you actually check if the sockets contain each 2 cores? IIUC, that's what the `'check_core_id_cmd'` branch should cover but as mentioned, that is never executed because the value is not defined (I executed this test and double checked in the debug.log).